Added wl.offline_access scope to support refresh token

jamesmcroft opened the pull request:
In order to keep a user logged into an application I'm working on, I required the refresh token to be returned as part of the authentication process. Added it to the Scopes enum. 

Possibly need to look into a different way of passing the scopes into the BexClient as some developers might want to add more Live specific scopes which would require adding them to the enum and going through this same cycle.

jamesmcroft committed:
Added the offline scope to retrieve a refresh token.

jamesmcroft committed:
Update README.md

ScottIsAFool commented:
How about changing it so that the BexClient accepts an IEnumerable<string> as well. The Scope enum can still be used but under the hood it converts it to IEnumerable<string>. Best of both worlds then.

jamesmcroft committed:
Updated the BexClient to accept a collection of scopes as their string representation to extend on the core health scopes exposed by Bex

jamesmcroft commented:
I've added a change which overloads the method used to create the authentication URL. I've removed the scope that I added to keep the enum clean with only the health specific scopes.

jamesmcroft committed:
Update README.md

ScottIsAFool commented:
Needs conflicts resolving.

jamesmcroft committed:
Merged upstream into master. Resolved merge conflict with IBexClient.

jamesmcroft commented:
My bad! I hadn't noticed you'd accepted another request which just so happens to do what I wanted to do in terms of the refresh token. I've merged everything in :+1:
